hooks/queries/useProgramVersionQuery.ts

```diff
@@ -0,0 +1,29 @@
+import { getGovernanceProgramVersion } from '@solana/spl-governance'
+import { PublicKey } from '@solana/web3.js'
+import { useQuery } from '@tanstack/react-query'
+import useWalletStore from 'stores/useWalletStore'
+
+export const programVersionQueryKeys = {
+  byProgramId: (programId: PublicKey) => [
+    'programVersion',
+    programId.toString(),
+  ],
+}
+
+export function useProgramVersionByIdQuery(realmsProgramId?: PublicKey) {
+  // @asktree is unsure why we use this instead of `useConnection` (which has no corresponding provider in the BaseApp)
+  const { connection } = useWalletStore()
+  console.log('connection', connection)
+
+  const query = useQuery({
+    queryKey:
+      realmsProgramId && programVersionQueryKeys.byProgramId(realmsProgramId),
+    queryFn: () =>
```

Comment:
the query itself is quite simple, since I'm just using an already-existing function `getGovernanceProgramVersion`. By essentially just wrapping it in a `useQuery` we get intelligent and easy caching, an easy api for knowing query status, etcetera.